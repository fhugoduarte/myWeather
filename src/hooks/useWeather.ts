import { useContextSelector } from "use-context-selector";

import { WeatherContext } from "~/providers/WeatherProvider";

export function useWeather() {
  const weather = useContextSelector(WeatherContext, (ctx) => ctx.weather);
  const loading = useContextSelector(WeatherContext, (ctx) => ctx.loading);
  const loadWeather = useContextSelector(
    WeatherContext,
    (ctx) => ctx.loadWeather
  );

  return { weather, loading, loadWeather };
}
