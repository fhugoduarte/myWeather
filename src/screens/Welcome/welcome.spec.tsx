import { render } from "@testing-library/react-native";
import { mocked } from "jest-mock";

import { useWeather } from "../../hooks/useWeather";
import { WelcomeScreen } from ".";

jest.mock("../../hooks/useWeather");

describe("Screens | Welcome", () => {
  it("should render correctly", () => {
    const useWeatherMocked = mocked(useWeather);

    useWeatherMocked.mockReturnValueOnce({
      weather: undefined,
    } as any);

    const tree = render(
      <WelcomeScreen navigation={{} as any} route={{} as any} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should redirect to WeatherForecastScreen when a weather is provided", () => {
    const navigateFunc = jest.fn();
    const useWeatherMocked = mocked(useWeather);

    useWeatherMocked.mockReturnValueOnce({
      weather: {},
    } as any);

    render(
      <WelcomeScreen
        navigation={{ navigate: navigateFunc } as any}
        route={{} as any}
      />
    );

    expect(navigateFunc).toHaveBeenCalledWith("WeatherForecast");
  });
});
