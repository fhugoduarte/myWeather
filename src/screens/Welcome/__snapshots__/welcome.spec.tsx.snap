// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens | Welcome should render correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4294950789,
        4294939525,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 47,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Inter_700Bold",
          "fontSize": 63,
        },
        Object {
          "textShadowColor": "#00000090",
          "textShadowOffset": Object {
            "height": 1,
            "width": -1,
          },
          "textShadowRadius": 3,
        },
        Object {
          "marginBottom": 24,
          "textAlign": "center",
        },
      ]
    }
  >
    Discover the Weather
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Inter_400Regular",
          "fontSize": 31,
        },
        Object {
          "textShadowColor": "#00000090",
          "textShadowOffset": Object {
            "height": 1,
            "width": -1,
          },
          "textShadowRadius": 3,
        },
        Object {
          "marginBottom": 63,
          "textAlign": "center",
        },
      ]
    }
  >
    Get to know the weather anywhere in the world
  </Text>
  <View
    accessibilityHint="Share your location and see the current weather"
    accessibilityLabel="Get the weather"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 35,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 63,
        "paddingVertical": 24,
      }
    }
  >
    <View
      accessible={false}
      importantForAccessibility="no"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#444E72",
              "fontFamily": "Inter_700Bold",
              "fontSize": 31,
            },
            Object {},
            undefined,
          ]
        }
      >
        Get the weather
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
